DSM/HE/HE_DSM_tindex.py

```diff
@@ -0,0 +1,336 @@
+geojson_dict = {
+  "type": "FeatureCollection",
+  "name": "tindex",
+  "crs": {"type": "name", "properties": {"name": "urn:ogc:def:crs:EPSG::25832"}},
+  "features": []
+}
+
+# find urls for each gem
+date = datetime.now().strftime("%Y%m%d")
+ALKIS_urls = {}
+krs_gem = {}
+krs_base_urls = {}
+num_HE_gem = len(vals)
+i = 0
+count = 0
+for cat, val in vals.items():
+    i += 1
+    if i % 10 == 0:
+        grass.message(_(f"{i}/{num_HE_gem} ..."))
+    krs_type = val[0]
+    krs = val[1]
+    gem = val[2]
+
+    # create url for krs and gem
+    print(type, krs, gem)
+    poss_krs_urls = create_urls(krs_type, krs, gem)
+
+    url = None
+    for poss_krs_url in set(poss_krs_urls):
+        if check_url(poss_krs_url):
+            url_today = poss_krs_url.replace("DATE", date)
+            tries = 0
+            zip = None
+            while tries <= MAX_RETRIES and zip is None:
+                try:
+                    zip = RemoteZip(url_today, timeout=1200)
+                except:
+                    sleep(15)
+                # grass.fatal(f"Error with RemoteZip for url <{url_today}> !")
+            zip_files = [zip_info.filename for zip_info in zip.infolist() if zip_info.filename.endswith(".tif")]
+            for zip_file in zip_files:
+                url = f"/vsizip/vsicurl/{poss_krs_url}/{zip_file}"
+                splitted_file_name = zip_file.split("_")
+                x1 = int(splitted_file_name[2]) * 1000
+                y1 = int(splitted_file_name[3]) * 1000
+                x2 = x1 + 1000
+                y2 = y1 + 1000
+                feat = {
+                    "type": "Feature", "properties": {
+                        "fid": count + 1,
+                        "location": url
+                    },
+                    "geometry": {
+                        "type": "Polygon", "coordinates": [[[x1, y1], [x2, y1], [x2, y2], [x1, y2], [x1, y1]]]
+                    }
+                }
+                geojson_dict["features"].append(feat)
+                count += 1
+    if url is None:
+        grass.fatal(f"No url found for {krs_type, krs, gem}!")
+
+"""CREATE TILE INDEX AS .gpkg.gz FILE"""
+
+# create GeoJson tile index
+geojson_name = "tindex.geojson"
+with open(geojson_name, "w") as f:
+    json.dump(geojson_dict, f, indent=4)
+
+# create GPKG from GeoJson
+gpkg_name = "HE_DSM_tindex.gpkg"
+stream = os.popen(f"ogr2ogr {gpkg_name} {geojson_name}")
+ogr2ogr_out = stream.read()
+
+# verify
+print("Verifying vector tile index:")
+stream = os.popen(f"ogrinfo -so -al {gpkg_name}")
+tindex_verification = stream.read()
+print(tindex_verification)
+
+# package
+if os.path.isfile(f"{gpkg_name}.gz"):
+    os.remove(f"{gpkg_name}.gz")
+stream = os.popen(f"gzip {gpkg_name}")
+create_gz = stream.read()
+print(f"<{gpkg_name}.gz> created")
+
+# cleanup
+if os.path.isdir("tmp"):
+    shutil.rmtree("tmp")
+if os.path.isfile(geojson_name):
+    os.remove(geojson_name)
+if os.path.isfile(gpkg_name):
+    os.remove(gpkg_name)
```

Comment:
maybe good to have the cleanup not only in the end, but also if script fails (e.g. with `atexit`).